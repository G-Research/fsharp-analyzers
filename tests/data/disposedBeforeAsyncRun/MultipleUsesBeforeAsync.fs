module M

open System

let asyncReturningFunc () =
    // should warn
    use t1 = new IO.FileStream("x", IO.FileMode.Open)
    // should warn
    use t2 = new IO.FileStream("x", IO.FileMode.Open)
    async {
        return "hi"
    }
