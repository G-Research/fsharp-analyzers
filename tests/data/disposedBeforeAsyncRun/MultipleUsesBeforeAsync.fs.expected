GRA-DISPBEFOREASYNC-001 | Warning | (7,8 - 7,10) | Object is disposed before returned async is run | []
GRA-DISPBEFOREASYNC-001 | Warning | (9,8 - 9,10) | Object is disposed before returned async is run | []
